### DISCUSSION TITLE
Editor input latency

### DISCUSSION THREAD

**Discussion by sicher** [helix-editor/helix#9442] [General]
I did some tests with Typometer (https://pavelfatin.com/typometer/) and got some interesting results. I'm running these on a macbook M1, macos Sonoma 14.2.1.

1. iTerm2 zsh. Average 11.5 ms latency.
2. iTerm2 zsh + vim. Average 13.5 ms latency.
3. iTerm2 zsh + helix. Average 20.7 ms latency (!!!!).

Note that this additional 9-10ms latency does not appear when running helix in Kitty nor in Alacritty. There I get a steady 20-21ms latency both with plain zsh and zsh+helix.

**Comment by kirawi** [comment]
What's your methodology? For example, barebones Vim does a lot less than Helix, Are you opening a file with syntax highlighting from tree-sitter? etc.

**Reply by archseer** [reply]
If you look at the post it's an iTerm2 specific thing and I think it has been reported before. Probably has something to do with either rendering sequences or event handling specific to that terminal.

**Reply by kirawi** [reply]
Yeah I saw that. I was just wondering if it had to do with syntax highlighting and how many events we'd be sending for that.

**Reply by sicher** [reply]
I start helix with an empty scratch buffer.

**Reply by sicher** [reply]
Oh, btw this is Helix 23.10.
